### ISSUE TITLE
Infinite recursion if type's base type is itself

### ISSUE THREAD

**Issue by HighOnMikey** [wsdl2phpgenerator/wsdl2phpgenerator#202]
I'm working with a web service which imports http://www.w3.org/2005/05/xmlmime. The base type for the types defined in this schema reference themselves. The struct returned from this particular service:

```
struct base64Binary {
 base64Binary _;
 anonymous0 contentType;
}
```

Due to this self-reference, [`getBaseTypeMembers()`](https://github.com/wsdl2phpgenerator/wsdl2phpgenerator/blob/6cb408acbc6dab52a8bb977d289d941dfba69ab2/src/ComplexType.php#L241) will call itself with infinite recursion.

Adding some quick checks resolves the issue:

```php
    protected function getBaseTypeMembers(ComplexType $type)
    {
        if (empty($type->baseType)) {
            return array();
        }

        if ($type === $type->baseType)
        {
            return array();
        }

        return array_merge($this->getBaseTypeMembers($type->baseType), $type->baseType->getMembers());
    }
```

However, the generated class for a self-referenced type will be extended by itself in [`generateClass()`](https://github.com/wsdl2phpgenerator/wsdl2phpgenerator/blob/6cb408acbc6dab52a8bb977d289d941dfba69ab2/src/ComplexType.php#L62). So, more checks are needed:

```php
    protected function generateClass()
    {

// ...

        $classBaseType = '';

        if ($this->baseType !== null && $this->baseType !== $this) {
            $classBaseType = $this->baseType->getPhpIdentifier();
        }

        $class = new PhpClass(
            $this->phpIdentifier,
            false,
            $classBaseType
        );

// ...
    }
```

**Comment by kasperg** [comment]
Thank you for reporting this @HighOnMikey.

I looked into the schema you reference as based on that it appears the the types to not reference themselves but define types in their own namespace which extend types in the `xs` namespace with the same names.

wsdl2phpgenerator has a poor understanding of namespaces due to the reliance on the PHP `SoapClient` which does not support namespaces (see also #136).

Based on your changes are the classes generated as expected regarding attributes types etc.?

**Comment by HighOnMikey** [comment]
Hey, sorry for the late response. I came to the same understanding as you regarding SOAP namespaces and PHP, but I can only determine at this time that the PR associated with this issue works only for my situation as I have not been able to test against multiple possibilities. For my situation, however, it does address the infinite recursion issue while correctly creating new filse/objects for `ComplexType` objects which reference themselves as their `baseType`.

**Comment by kasperg** [comment]
Thanks for following up @HighOnMikey.

Let's continue the discussion in #203.

**Comment by kasperg** [comment]
This is now fixed in 3.2.0.
